<!-- Run the aruco_detect node -->
<launch>
  <!-- namespace for camera input -->
    <arg name="camera_1" default="logitech" />
    <arg name="camera_2" default="usb_cam_2" />

    <arg name="camera_cfg" default="logitech_480.yaml"/>


    <arg name="device_cam_1" default="2"/>
    <arg name="device_cam_2" default="4"/>

    <arg name="image" default="image_raw" />
    <arg name="transport" default="compressed" />
    <arg name="do_pose_estimation" default="true"/>
    <!-- If vis_msgs set to true, pose estimation will be published with ROS standard vision_msgs -->
    <arg name="vis_msgs" default="false"/>
    <arg name="ignore_fiducials" default=""/>
    <arg name="fiducial_len_override" default=""/>
    <arg name="verbose" default="false"/>

    <arg name="fiducial_len" default="0.055"/>
    <arg name="dictionary" default="0"/>

    <arg name="width" default="640"/>
    <arg name="height" default="480"/>

    <!-- <arg name="width" default="1280"/>
    <arg name="height" default="720"/> -->
    
    <arg name="num_fiducials" default="3"/>
    <!-- <arg name="camera_cfg_2" default="logitech.yaml"/> -->

    <node pkg="usb_cam" type="usb_cam_node" name="$(arg camera_1)" output="screen" >
        <param name="video_device" value="/dev/video$(arg device_cam_1)"/>
        <param name="image_width" value="$(arg width)"/>
        <param name="image_height" value="$(arg height)"/>
        <param name="pixel_format" value="yuyv"/>
        <param name="camera_frame_id" value="usb_cam"/>
        <param name="io_method" value="mmap"/>
        <param name="camera_info_url" value="file://$(find usb_cam)/calibration_files/$(arg camera_cfg)"/>
        <param name="camera_name" value="$(arg camera_1)"/>
        <!-- <param name="framerate" value="30"/> -->
        
    </node>

    <!-- <node pkg="usb_cam" type="usb_cam_node" name="$(arg camera_2)" output="screen" >
        <param name="video_device" value="/dev/video$(arg device_cam_2)"/>
        <param name="image_width" value="$(arg width)"/>
        <param name="image_height" value="$(arg height)"/>
        <param name="pixel_format" value="yuyv"/>
        <param name="camera_frame_id" value="usb_cam"/>
        <param name="io_method" value="mmap"/>
        <param name="camera_info_url" value="file://$(find usb_cam)/calibration_files/$(arg camera_cfg)"/>
        <param name="camera_name" value="webcam"/>
    </node> -->


    <node pkg="aruco_detect" name="aruco_detect" type="aruco_detect" output="screen" respawn="false">
        <param name="image_transport" value="$(arg transport)"/>
        <param name="publish_images" value="true" />
        <param name="fiducial_len" value="$(arg fiducial_len)"/>
        <param name="dictionary" value="$(arg dictionary)"/>
        <param name="do_pose_estimation" value="$(arg do_pose_estimation)"/>
        <param name="vis_msgs" value="$(arg vis_msgs)"/>
        <param name="ignore_fiducials" value="$(arg ignore_fiducials)"/>
        <param name="fiducial_len_override" value="$(arg fiducial_len_override)"/>
        <param name="verbose" value="$(arg verbose)"/>
        <remap from="camera/compressed" 
            to="/$(arg camera_1)/$(arg image)/$(arg transport)"/>
        <remap from="camera_info" to="/$(arg camera_1)/camera_info"/>
    </node>


    <node pkg="uav_positioning" type="camera_node" name="cam_1" output="screen">
        <param name="num_fiducials" value="$(arg num_fiducials)" />
        <param name="dx_camera" value="0.0"/>
        <param name="dy_camera" value="0.0"/>
        <param name="dz_camera" value="0.0"/>
        <param name="rotation_camera" value="0.0"/>
    </node> 

    <!-- n... camaras -->

    <!-- <node pkg="uav_positioning" type="reduce_cameras" name="reduction" output="screen"> </node> -->
    
    <node pkg="image_view" type="image_view" name="image_view" respawn="false" output="screen">
        <remap from="image" to="/fiducial_images"/>
        <param name="autosize" value="true"/>
    </node>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find uav_positioning)/launch/config.rviz"/>

</launch>
